ast/ddl.go

```diff
@@ -71,6 +84,21 @@ type CreateDatabaseStmt struct {
 	Options     []*DatabaseOption
 }
 
+// Restore implements Recoverable interface.
+func (n *CreateDatabaseStmt) Restore(sb *strings.Builder) {
+	sb.WriteString("CREATE DATABASE ")
+	if n.IfNotExists {
+		sb.WriteString("IF NOT EXISTS ")
+	}
+	sb.WriteString("`")
+	sb.WriteString(n.Name)
+	sb.WriteString("`")
```

Comment:
1. Will it properly handle names with `` ` `` in it e.g.:

    ```sql
    CREATE DATABASE `foo``bar`;
    ```

2. Do we want to handle ANSI_QUOTES (which the syntax becomes ``CREATE DATABASE "foo`bar"``) or is it considered out-of-scope?

Reply:
@leoppro In `tidb-tools` we have https://github.com/pingcap/tidb-tools/blob/c106fbb7c074df812df90e9035a6dc59a1853565/pkg/dbutil/common.go#L396 to escape the backquotes, but I'm not sure if `n.Name` will unescape the backquotes in the first place.

For MySQL even if ANSI_QUOTES is enabled the `` `...` `` syntax can still be used, so no function is lost if we ignore ANSI_QUOTES. It's just a question of aesthetic.

Reply:
what about now?
https://github.com/pingcap/parser/pull/56/files#diff-ed076aa6b0e8db7b5db0c844dcc1cb33R5